File: conference/payments-bc/web/src/main/java/com/microsoft/conference/payments/domain/Models/Payment.cs
using System;
using System.Collections.Generic;
using ENode.Domain;

namespace Payments
{
    public class Payment : AggregateRoot<Guid>
    {
        private Guid _orderId;
        private Guid _conferenceId;
        private PaymentState _state;
        private string _description;
        private decimal _totalAmount;
        private IEnumerable<PaymentItem> _items;

        public Payment(Guid id, Guid orderId, Guid conferenceId, string description, decimal totalAmount, IEnumerable<PaymentItem> items) : base(id)
        {
            ApplyEvent(new PaymentInitiated(orderId, conferenceId, description, totalAmount, items));
        }

        public void Complete()
        {
            if (_state != PaymentState.Initiated)
            {
                throw new InvalidOperationException();
            }
            ApplyEvent(new PaymentCompleted(this, _orderId, _conferenceId));
        }
        public void Cancel()
        {
            if (_state != PaymentState.Initiated)
            {
                throw new InvalidOperationException();
            }
            ApplyEvent(new PaymentRejected(_orderId, _conferenceId));
        }

        private void Handle(PaymentInitiated evnt)
        {
            _id = evnt.AggregateRootId;
            _orderId = evnt.OrderId;
            _conferenceId = evnt.ConferenceId;
            _description = evnt.Description;
            _totalAmount = evnt.TotalAmount;
            _state = PaymentState.Initiated;
            _items = evnt.Items;
        }
        private void Handle(PaymentCompleted evnt)
        {
            _state = PaymentState.Completed;
        }
        private void Handle(PaymentRejected evnt)
        {
            _state = PaymentState.Rejected;
        }
    }
    public class PaymentItem
    {
        public PaymentItem(string description, decimal amount)
        {
            this.Id = Guid.NewGuid();
            this.Description = description;
            this.Amount = amount;
        }

        public Guid Id { get; private set; }
        public string Description { get; private set; }
        public decimal Amount { get; private set; }
    }
    public enum PaymentState
    {
        Initiated = 0,
        Completed = 1,
        Rejected = 2
    }
}

File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/domain/models/Payment.java
package com.microsoft.conference.payments.domain.models;

import com.microsoft.conference.common.exception.InvalidOperationException;
import com.microsoft.conference.payments.domain.events.PaymentCompleted;
import com.microsoft.conference.payments.domain.events.PaymentInitiated;
import com.microsoft.conference.payments.domain.events.PaymentRejected;
import org.enodeframework.domain.AggregateRoot;

import java.math.BigDecimal;
import java.util.List;

public class Payment extends AggregateRoot<String> {
    private String orderId;
    private String conferenceId;
    private int state;
    private String description;
    private BigDecimal totalAmount;
    private List<PaymentItem> paymentItems;

    public Payment() {
    }

    public Payment(String id, String orderId, String conferenceId, String description, BigDecimal totalAmount, List<PaymentItem> items) {
        super(id);
        applyEvent(new PaymentInitiated(orderId, conferenceId, description, totalAmount, items));
    }

    public void complete() {
        if (state != PaymentState.Initiated) {
            throw new InvalidOperationException();
        }
        applyEvent(new PaymentCompleted(this, orderId, conferenceId));
    }

    public void cancel() {
        if (state != PaymentState.Initiated) {
            throw new InvalidOperationException();
        }
        applyEvent(new PaymentRejected(orderId, conferenceId));
    }

    private void handle(PaymentInitiated evnt) {
        id = evnt.getAggregateRootId();
        orderId = evnt.getOrderId();
        conferenceId = evnt.getConferenceId();
        description = evnt.getDescription();
        totalAmount = evnt.getTotalAmount();
        state = PaymentState.Initiated;
        paymentItems = evnt.getPaymentItems();
    }

    private void handle(PaymentCompleted evnt) {
        state = PaymentState.Completed;
    }

    private void handle(PaymentRejected evnt) {
        state = PaymentState.Rejected;
    }
}
